task_02/ex5.py

```diff
@@ -0,0 +1,30 @@
+"""The Most Frequent."""
+
+# https://py.checkio.org/en/mission/the-most-frequent/
+
+
+def most_frequent(list_arg):
+    """You have a sequence of strings.
+
+    And you’d like to determine
+    the most frequently occurring string in the sequence.
+    """
+    if type(list_arg) is not list:
```

Comment:
How can't it be not a list? `split()` will return either list or raise exception itself. If you suppose your function can be called from other places, why can't we send a tuple? It would be OK for tuples.

Reply:
List is a mutable type, whilst tuple is not. First, I do not know where my function can be implemented further, nevertheless even now I consider a situation, when someone, using my function, is lack of memory or something this way. Sure, it seems suspicious and hardly probable, though this situation may happen. Moreover, it's the first time I write such exceptions in functions. I mean cpp/c#/java implement data types, while python doesn't. It really makes me feel inconvienient and suspicious. Second, list as and input argument is the condition of the exercise, thus I implemented it explicitely.

Reply:
I see sequence, not list in task conditions. 
Python is a strong type language, but it's very flexible, that's why we love him. 
To make some type checking isn't pythonic. we prefer to write without them. Our functions are more flexible without such checks.
>
> I consider a situation, when someone, using my function, is lack of memory or something this wa...
>
You doesn't change list in your code. It wouldn't have bigger memory consumptions with tuples.  More of that as I know tuple is more lightweight objects than lists are. 
Python seems to be a big proxy between pythonist and os functions, thats why we don't think about such things by default. We will talk about it during our lessons concerning profiling and so on.